docs/topics/settings.rst

```diff
@@ -28,6 +28,11 @@ The value of ``SCRAPY_SETTINGS_MODULE`` should be in Python path syntax, e.g.
 ``myproject.settings``. Note that the settings module should be on the
 Python :ref:`import search path <tut-searchpath>`.
 
+.. note::
+    Configuring settings of python types that are not compatible with the
+    `pickle <https://github.com/python/cpython/blob/3.10/Lib/pickle.py>`_
+    module will cause Scrapy to fail.
```

Comment:
I think that, rather than a note on an unrelated section, we should give this a section of its own, e.g. between “Populating the settings” and “Import paths and classes” (which I have just realized should be called “Import paths and callables”, but that is out of the scope of this change).

Also, we should:

- Link to the module using InterSphinx: `` :mod:`pickle` ``.
- Include a link to https://docs.python.org/3/library/pickle.html#what-can-be-pickled-and-unpickled, also using InterSphinx: `` :ref:`pickle-picklable` ``

Reply:
Made commit [e920db2](https://github.com/scrapy/scrapy/pull/5629/commits/e920db27954e66506798d57620c0a45e6f567cde):
- Added this information in its own section
- Linked to module using InterSphinx
- Linked to relevant section of docs using InterSphinx